uvicorn/_handlers/http.py

```diff
@@ -0,0 +1,36 @@
+import asyncio
+
+from uvicorn.config import Config
+from uvicorn.server import ServerState
+
+MAX_RECV = 65536
+
+
+async def handle_http(
+    reader: asyncio.StreamReader,
+    writer: asyncio.StreamWriter,
+    server_state: ServerState,
+    config: Config,
+) -> None:
+    # Run transport/protocol session from streams.
+    # https://docs.python.org/3/library/asyncio-protocol.html#connecting-existing-sockets
+
+    loop = asyncio.get_event_loop()
+    connection_lost = loop.create_future()
+
+    protocol = config.http_protocol_class(
+        config=config,
+        server_state=server_state,
+        on_connection_lost=lambda: connection_lost.set_result(True),
+    )
+    transport = writer.transport
+    transport.set_protocol(protocol)
+
+    # Kick off the HTTP protocol, passing any data already in the read buffer.
+    protocol.connection_made(transport)
```

Comment:
```suggestion
    protocol.connection_made(transport)
    await writer.drain()
```

Reply:
Apologies if this sounds stupid, it's been a while I havnt played with this kind of stuff, but iirc shouldn't we drain the writer before we read data ?

Reply:
Don't think so? The write buffer is related to… us sending data to the client. At this point in time we haven't written anything yet, so I don't think there's a need to `.drain()`.

Reply:
ok I didnt get it was this early in the setup